### PR TITLE
Implement `rev`

### PR DESCRIPTION
**Pull request by sisungo**

### CODE REVIEW

File: src/uu/rev/src/rev.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    let mut has_error = false;
+
+    match files {
+        Some(files) => {
+            for path in files {
+                let file = match std::fs::File::open(path) {
+                    Ok(val) => val,
+                    Err(err) => {
+                        has_error = true;
+                        eprintln!("rev: cannot open {}: {}", path, err);
```
Comment (cakebaker) [inline src/uu/rev/src/rev.rs:28]: You can simplify the error handling by using either `show!` or `show_error!` from `uucore`.

File: src/uu/rev/src/rev.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                        continue;
+                    }
+                };
+                if let Err(err) = rev_stream(file) {
+                    has_error = true;
+                    eprintln!("rev: cannot process {}: {}", path, err);
```
Comment (cakebaker) [inline src/uu/rev/src/rev.rs:34]: See previous comment.

File: src/uu/rev/src/rev.rs (hunk 1 above)
Comment (sisungo) [reply src/uu/rev/src/rev.rs]: The `eprintln!`s are replaced by `show_error!` in latest commit, thanks.

For `has_error`, I observed `util-linux`'s `rev`'s behavior is:

 * If any error occurred, it immediately prints the error, but not terminating the process, instead, it continues to process files passed from command-line arguments;
 * If any error occurred, the final exit code will be `1`.
 * If all files are `rev`-ed successfully, the final exit code will be `0`.

That's why `has_error` was set.

File: src/uu/rev/src/rev.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    let files = matches.get_many::<String>("file");
+    let mut has_error = false;
+
+    match files {
+        Some(files) => {
+            for path in files {
+                let file = match std::fs::File::open(path) {
+                    Ok(val) => val,
+                    Err(err) => {
+                        has_error = true;
```
Comment (cakebaker) [inline src/uu/rev/src/rev.rs:27]: You can further improve the code and getting rid of the `has_error` variable (and the match at the end of the function) by using `set_exit_code` from `uucore`.

**Comment by sisungo** [comment]
Thanks for giving help! Now I've removed use of `has_error` variable and switched to `uucore`'s error handling facility.

**Comment by cakebaker** [comment]
Thanks for your PR :)

**Pull request merged**